execution/executor-benchmark/src/db_access.rs

```diff
@@ -0,0 +1,154 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use aptos_state_view::TStateView;
+use aptos_storage_interface::{cached_state_view::CachedStateView, state_view::DbStateView};
+use aptos_types::{
+    access_path::AccessPath, account_address::AccountAddress, state_store::state_key::StateKey,
+};
+use move_core_types::{
+    identifier::Identifier,
+    language_storage::{StructTag, TypeTag},
+};
+use once_cell::sync::Lazy;
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use std::str::FromStr;
+
+// Note: in case this changes in the future, it doesn't have to be a constant, and can be read from
+// genesis directly if necessary.
+pub static TOTAL_SUPPLY_STATE_KEY: Lazy<StateKey> = Lazy::new(|| {
+    StateKey::table_item(
```

Comment:
Scary 😨

Reply:
@grao1991's code, just extracted to a separate file here :)